src/blocks/textInput/TextInput.tsx

```diff
@@ -0,0 +1,191 @@
+  /* Label for the input field */
+  label?: string;
+  /* TotalLength of input value */
+  totalCount?: number;
+  /* Placeholder for input field */
+  placeholder?: string;
+  /* Sets the input field to be compulsory */
+  required?: boolean;
+  /* Sets the input field to error state */
+  error?: boolean;
+  /* Sets the input field to success state */
+  success?: boolean;
+  /* Sets button as disabled */
+  disabled?: boolean;
+  /* Description shown below the input field */
+  description?: string;
+  /* Error message shown below the input field */
+  errorMessage?: string;
+};
+
+const StyledTextInput = styled.div<TextInputProps & ModeProp>`
+  /* Common Input field CSS */
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  gap: var(--s2);
+  .label-count {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  }
+
+  .input-container {
+    cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
+    display: flex;
+    align-items: center;
+    font-family: var(--font-family);
+    font-size: 14px;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 20px;
+    white-space: nowrap;
+    padding: var(--s0) var(--s3);
+    border-radius: var(--r3);
+
+    /* Common icon css added through CSS class */
+    .icon {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      margin-right: 2px;
+      span {
+        width: 18px;
+        height: 13.5px;
+      }
+    }
+
+    & input {
+      border: none;
+      background-color: transparent;
+      width: 100%;
+      padding: var(--s3) var(--s0);
+      &:focus {
+        outline: none;
+      }
+      &:hover {
+        outline: none;
+      }
+      :disabled {
+        background-color: transparent;
+      }
+    }
+  }
+  /* TextInput type CSS styles */
+  ${({ mode, error, disabled }) =>
+    getTextInputStateStyles({ mode, state: getTextInputState({ error: !!error, disabled: !!disabled }) })}
+
+  /* Custom CSS applied via styled component css prop */
+  ${(props) => props.css || ''}
+`;
+
+const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+  (
+    {
+      disabled,
+      error,
+      success,
+      required,
+      label,
+      totalCount,
+      placeholder,
+      icon,
+      type = 'text',
+      onChange,
+      value,
+      onClear,
+      description,
+      errorMessage,
+
+      ...props
+    },
+    ref
+  ) => {
+    const { mode } = useBlocksTheme();
+    const state = getTextInputState({ error: !!error, disabled: !!disabled });
+    return (
+      <StyledTextInput
+        disabled={disabled}
+        error={error}
+        required={required}
+        role="input"
+        mode={mode}
+        ref={ref}
+        {...props}
+      >
+        {(label || totalCount) && (
+          <div className="label-count">
+            {label && <Text className="label">{label}</Text>}
+            {totalCount && <Text className="count">{totalCount}</Text>}
```

Comment:
```suggestion
            {label && <Text className="label">{label}</Text>}
            {totalCount && <Text className="count">{totalCount}</Text>}
```

For text the styling should not come from className but from the variant prop. If the variant is not available please ask zee to provide it and then we also add it in Text.

Reply:
removing the class name but the styling for the same isnt done in this pr , will do it in the later ticket for textInput